packages/web-wallet/src/dataProviders/credentialDataProvider.ts

```diff
@@ -0,0 +1,154 @@
+          filterItem[filterField] = filter.value
+        }
+      } else {
+        throw new Error(`operator ${filter.operator} not yet supported`)
+      }
+    })
+
+    const fetchOptions: FetchOptions = {}
+    if (meta?.variables && 'showVersionHistory' in meta.variables) {
+      fetchOptions.showVersionHistory = meta.variables.showVersionHistory
+    }
+
+    const items: Array<DigitalCredential> = await agent.crsGetCredentials({filter: findArgs})
+    // FIXME CWALL-234 there should be a better way for this but i could not find any yet without refine.dev not complaining
+    const data: TData[] = items.map(item => ({...(item as any)}))
+
+    return {
+      data,
+      total: items.length,
+    }
+  },
+  getOne: async <TData extends BaseRecord = BaseRecord>({resource, id, meta}: GetOneParams): Promise<GetOneResponse<TData>> => {
+    assertResource(resource)
+    console.log('== meta', meta)
+    if (meta === undefined || meta.variables === undefined || !('credentialRole' in meta.variables)) {
+      return Promise.reject(Error('credentialRole not found in meta query'))
+    }
+    const args: GetCredentialsByIdOrHashArgs = {credentialRole: meta.variables.credentialRole, idOrHash: id as string}
+    const credential: OptionalUniqueDigitalCredential = await agent.crsGetUniqueCredentialByIdOrHash(args)
+    return {
+      data: credential?.digitalCredential as unknown as TData,
+    }
+  },
+  create: async <TData extends BaseRecord = BaseRecord, TVariables = {}>({
+    resource,
+    variables,
+  }: CreateParams<TVariables>): Promise<CreateResponse<TData>> => {
+    assertResource(resource)
+    const item: DigitalCredential = await agent.crsAddCredential({credential: variables as DigitalCredential})
+    return {
+      data: item as unknown as TData,
+    }
+  },
+  update: async <TData extends BaseRecord = BaseRecord, TVariables = {}>({
+    resource,
+    id,
+    variables,
+  }: UpdateParams<TVariables>): Promise<UpdateResponse<TData>> => {
+    assertResource(resource)
+    const item: DigitalCredential = await agent.crsUpdateCredentialState(variables as UpdateCredentialStateArgs)
+    return {
+      data: item as unknown as TData,
+    }
+  },
+  deleteOne: async <TData extends BaseRecord = BaseRecord, TVariables = {}>({
+    resource,
+    id,
+  }: DeleteOneParams<TVariables>): Promise<DeleteOneResponse<TData>> => {
+    assertResource(resource)
+    await agent.crsDeleteCredential({id: id as string})
+    return {
+      data: {} as TData,
+    }
+  },
+  deleteMany: async <TData, TVariables>(params: DeleteManyParams<TVariables>): Promise<DeleteManyResponse<TData>> => {
+    assertResource(params.resource)
+    const filter: FindDigitalCredentialArgs = params.ids.map(id => {
+      return {id: id as string}
+    })
+    await agent.crsDeleteCredentials({
+      filter: filter,
+    })
+    return {
+      data: [] as TData[],
+    }
+  },
+
+  getApiUrl: (): string => {
+    // TODO implement
+    return 'some api url'
```

Comment:
Throw an error

Reply:
ok
Fixed it everywhere
![image](https://github.com/user-attachments/assets/2e7b7b8b-07b3-45c6-967e-45aa7ff05d4f)